repobee_sanitizer/_sanitize_repo.py

```diff
@@ -92,10 +92,38 @@ def sanitize_files(
     return []
 
 
+def create_pr_branch(
+    repo_path: pathlib.Path, target_branch: str, pr_branch_name: str
+):
+    """Create a new branch from the target_branch that we sanitize to
+    and a pull request can be created from.
+
+    Args:
+        repo_path: Path to the repository
+        target_branch: The path to create the pull request branch from
+        pr_branch_name: The name of the branch to sanitize to and create a pr
+        from
+    """
+
+    repo = git.Repo(str(repo_path))
+    source_branch = repo.head.ref.path
+
+    # Assures the target branch is not empty
+    try:
+        repo.git.symbolic_ref("HEAD", f"refs/heads/{target_branch}")
+        repo.index.diff("HEAD")
+    except git.BadName:
+        repo.git.commit("--allow-empty", "-m", "initial commit")
```

Comment:
Several problems here:

1. Avoid using plumbing commands like `symbolic-ref` whenever possible. They are extremely error prone. It's much better to create branches either with `branch` or `checkout`.
2. Like I've already mentioned (and you haven't yet responded to, so please do!), I do not see any good reason to allow one to create a pull request branch into a target branch that does not exist by creating an empty target branch. Why would I as a user want that to happen? To add insult to injury, none of the documentation or anything in the CLI help section indicates that this happens. Could you provide some reasoning for why it works like this? A user story where it's useful?
2. This is a pretty nasty hack to ensure that a branch exists, it's very much non-obvious from reading the code what `repo.index.diff` has to do with anything.

On a side note, the vast majority of Git commands do not require you to specify `"HEAD"`. This is in fact the very point of having the `HEAD` reference, to let commands such as `diff`, `commit` etc know what is the current commit.

Reply:
Yes! I agree that making sure a branch is not empty is silly and i plan on changing it :)

Reply:
Now i remember why i wanted this, when using solutions-driven development, you might start out on the solutions branch, and then have your master or whatever is your target branch be empty since you havnt done anything there, and i didnt like the idea of forcing the user to commit something "just because".

Also documentation comes when were finished :)

Reply:
> Now i remember why i wanted this, when using solutions-driven development, you might start out on the solutions branch, and then have your master or whatever is your target branch be empty since you havnt done anything there, and i didnt like the idea of forcing the user to commit something "just because".

It still doesn't make sense to create a pull request into a branch that does not exist. The user should simply make the first commit  on the target branch as a "normal" sanitization to that target branch, which then creates the branch _with_ some meaningful content. Note that it works fine to use `--target-branch` with a target branch that doesn't actually exist yet.

Your solution here just creates the PR branch, and a target branch with an empty commit. That is pointless, as the target branch in question won't exist on the remote platform anyway. The user would have to push _both_ branches before being able to create the PR. Why then not just sanitize directly to the target branch, and push that? 

The whole motivation to allow for a pull request workflow is to allow the user to lock the target branch on the platform, and then be able to inspect every PR before merging. But if the target branch _does not exist_ on the platform, there is nothing to lock, and there are no changes to inspect.

> Also documentation comes when were finished :)

When you request a review, I assume that the PR is finished. If it isn't, then please ask for specific feedback so I don't write unnecessary feedback on stuff you already know you haven't done.